### PR TITLE
Ajax viewer for arks in bibid

### PR DESCRIPTION
**Pull request by christinach**

### CODE REVIEW

**Comment by christinach** [comment]
Updates the viewer.
Displays 'Digital Content' in the 'show' and 'search results' page.

Co-authored-by: Nikitas Tampakis nikitas.tampakis@gmail.com

**Comment by christinach** [comment]
Closes #1281
Closes #1280
Closes #1211

**Comment by christinach** [comment]
Adds an Ajax call to figgy for bibids with arks and no iif_manifest paths. If none of them are available adds a "Digital content is not available."
For viewers generated through the ajax call, it supports navigation to the viewer/s through the links in the 'Online' panel.
For viewers generated through the iif_manifest paths it supports navigation to the first viewer. This will get updated with pulibrary/marc_liberation#415.

**Comment by christinach** [comment]
Closes #1272

**Comment by christinach** [comment]
I tested the XMLHttpRequest manually locally and in staging. I was not able to fire it from the spec.

File: lib/tasks/pulsearch.rake [hunk 1]
```diff
@@ -48,6 +48,7 @@ namespace :pulsearch do
   private
 
     def url_for_file(file)
-      "https://raw.githubusercontent.com/pulibrary/pul_solr/master/solr_configs/orangelight/#{file}"
+      #  "https://raw.githubusercontent.com/pulibrary/pul_solr/master/solr_configs/orangelight/#{file}"
+      "https://raw.githubusercontent.com/pulibrary/pul_solr/cf686290a6e191987e43f5462873f1556763d438/solr_configs/orangelight/#{file}"
```
Comment (tpendragon) [inline lib/tasks/pulsearch.rake]: Why attached to a specific commit?

Comment (christinach) [reply lib/tasks/pulsearch.rake]: This line will be updated as soon as @tampakis updates the orangelight code to work with the new solr changes. Right now the production orangelight is still using the old configs (from the specific commit).

Comment (christinach) [reply lib/tasks/pulsearch.rake]: @tpendragon I removed the specific commit to avoid confusion. The test will pass as soon as we update the solr configs. Thanks for your comment!

**Comment by christinach** [comment]
@jrgriffiniii thank you for pointing out this method. The PR will break in one for the ajax figgy viewer and one for the 'Digital Content'. @tampakis will handle this method in a separate PR for the Digital Content.

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -46,6 +46,17 @@ def search_links(electronic_access)
...
+  def electronic_access_link(url, texts)
+    markup = if /Open access/.match? texts.first
+               link_to(texts.first, url.to_s, target: '_blank')
+             elsif %r{(\/catalog\/.+?#view)} =~ url.to_s
+               link_to('Digital content', $&)
+             else
+               link_to(texts.first, "#{ENV['proxy_base']}#{url}", target: '_blank')
+             end
+    markup
+  end
```
Comment (christinach) [inline app/helpers/application_helper.rb]: move to a separate PR.

**Comment by tampakis** [comment]
add yarn-error.log to the .gitignore

File: spec/views/catalog/show.html.erb_spec.rb [hunk 3]
```diff
@@ -7,14 +7,50 @@
...
 
-  it 'renders more than one IIIF viewers if they exist' do
+  it 'renders more than one IIIF viewers if the iiif_manifest_paths exist' do
     visit '/catalog/3943643'
     expect(page).to have_selector('div#view')
     expect(page).to have_selector('div#view_1')
   end
 
-  it 'renders one viewer for one ark' do
+  it 'renders one viewer for one iif_manifest_path ark' do
```
Comment (tampakis) [inline spec/views/catalog/show.html.erb_spec.rb]: typo iiif_manifest_path

**Comment by tampakis** [comment]
🎉
